perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -405,7 +409,15 @@ def WriteObjectFromBuffer(storage_schema, bucket_name, object_name,
     if host_to_connect is not None:
       object_uri.connect(host=host_to_connect)
 
-    object_uri.set_contents_from_string(payload_string)
+    if FLAGS.storage_class and storage_schema == 's3':
+      # We need to use a lower-level upload method when using an S3
+      # storage class because we need access to the key object so we
+      # can set its storage class.
+      key = object_uri.new_key()
+      key._set_storage_class(FLAGS.storage_class)
+      key.set_contents_from_string(payload_string)
```

Comment:
Please merge in the other PR you did that changed this function from calling set_contents_from_string() to set_contents_from_file().